### ISSUE TITLE
Additional filter on saved question results in hanging "Doing Science" dialog, doesn't redraw.

### ISSUE THREAD

**Issue by mattxbart** [metabase/metabase#3687]
Might be related to #3146. I have a question saved on a dashboard. When I click into the table to do an additional filter or refine the question, the "Doing Science" dialog comes up but never exits. I can see the call being made to /api/dataset and returning status 200 along with the data but the dialog never exits or redraws the screen. When I cancel and refresh the data appears. Seems to happen pretty consistently, the query isn't very complex, runs quickly in my dbshell and on the dashboard. 

Metabase Version: 0.20.3 & 0.20.0
Browser: Chrome
OS: Ubuntu 16
Database: Postgres

**Comment by mattxbart** [comment]
I'm seeing this JavaScript error, I also tried reducing the number of columns in my table that didn't seem to help:

```
crossfilter.js:462 Uncaught (in promise) Error: invalid array width!
    at A (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:129:31107)
    at Object.i [as dimension] (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:129:29566)
    at s (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:16:13238)
    at v (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:16:13358)
    at b (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:16:13530)
    at Function.value (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:41:6419)
    at mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:118:10973
    at Array.map (native)
    at t.value (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:118:10708)
    at t.value (mydomain/app/dist/app.bundle.js?9e2c934c2d9d09b08aca:118:11389)A @ crossfilter.js:462i @ crossfilter.js:635s @ utils.js:192v @ utils.js:190b @ utils.js:198value @ LineAreaBarChart.jsx:30(anonymous function) @ VisualizationSettings.jsx:61value @ VisualizationSettings.jsx:56value @ VisualizationSettings.jsx:80_renderValidatedComponentWithoutOwnerOrContext @ ReactCompositeComponent.js:796_renderValidatedComponent @ ReactCompositeComponent.js:819_updateRenderedComponent @ ReactCompositeComponent.js:743_performComponentUpdate @ ReactCompositeComponent.js:721updateComponent @ ReactCompositeComponent.js:642receiveComponent @ ReactCompositeComponent.js:544receiveComponent @ ReactReconciler.js:126updateChildren @ ReactChildReconciler.js:110_reconcilerUpdateChildren @ ReactMultiChild.js:215_updateChildren @ ReactMultiChild.js:314updateChildren @ ReactMultiChild.js:301_updateDOMChildren @ ReactDOMComponent.js:942updateComponent @ ReactDOMComponent.js:760receiveComponent @ ReactDOMComponent.js:718receiveComponent @ ReactReconciler.js:126updateChildren @ ReactChildReconciler.js:110_reconcilerUpdateChildren @ ReactMultiChild.js:215_updateChildren @ ReactMultiChild.js:314updateChildren @ ReactMultiChild.js:301_updateDOMChildren @ ReactDOMComponent.js:942updateComponent @ ReactDOMComponent.js:760receiveComponent @ ReactDOMComponent.js:718receiveComponent @ ReactReconciler.js:126updateChildren @ ReactChildReconciler.js:110_reconcilerUpdateChildren @ ReactMultiChild.js:215_updateChildren @ ReactMultiChild.js:314updateChildren @ ReactMultiChild.js:301_updateDOMChildren @ ReactDOMComponent.js:942updateComponent @ ReactDOMComponent.js:760receiveComponent @ ReactDOMComponent.js:718receiveComponent @ ReactReconciler.js:126_updateRenderedComponent @ ReactCompositeComponent.js:751_performComponentUpdate @ ReactCompositeComponent.js:721updateComponent @ ReactCompositeComponent.js:642receiveComponent @ ReactCompositeComponent.js:544receiveComponent @ ReactReconciler.js:126updateChildren @ ReactChildReconciler.js:110_reconcilerUpdateChildren @ ReactMultiChild.js:215_updateChildren @ ReactMultiChild.js:314updateChildren @ ReactMultiChild.js:301_updateDOMChildren @ ReactDOMComponent.js:942updateComponent @ ReactDOMComponent.js:760receiveComponent @ ReactDOMComponent.js:718receiveComponent @ ReactReconciler.js:126updateChildren @ ReactChildReconciler.js:110_reconcilerUpdateChildren @ ReactMultiChild.js:215_updateChildren @ ReactMultiChild.js:314updateChildren @ ReactMultiChild.js:301_updateDOMChildren @ ReactDOMComponent.js:942updateComponent @ ReactDOMComponent.js:760receiveComponent @ ReactDOMComponent.js:718receiveComponent @ ReactReconciler.js:126updateChildren @ ReactChildReconciler.js:110_reconcilerUpdateChildren @ ReactMultiChild.js:215_updateChildren @ ReactMultiChild.js:314updateChildren @ ReactMultiChild.js:301_updateDOMChildren @ ReactDOMComponent.js:942updateComponent @ ReactDOMComponent.js:760receiveComponent @ ReactDOMComponent.js:718receiveComponent @ ReactReconciler.js:126_updateRenderedComponent @ ReactCompositeComponent.js:751_performComponentUpdate @ ReactCompositeComponent.js:721updateComponent @ ReactCompositeComponent.js:642receiveComponent @ ReactCompositeComponent.js:544receiveComponent @ ReactReconciler.js:126_updateRenderedComponent @ ReactCompositeComponent.js:751_performComponentUpdate @ ReactCompositeComponent.js:721updateComponent @ ReactCompositeComponent.js:642performUpdateIfNecessary @ ReactCompositeComponent.js:558performUpdateIfNecessary @ ReactReconciler.js:158s @ ReactUpdates.js:151perform @ Transaction.js:138perform @ Transaction.js:138perform @ ReactUpdates.js:90T @ ReactUpdates.js:173closeAll @ Transaction.js:204perform @ Transaction.js:151batchedUpdates @ ReactDefaultBatchingStrategy.js:63l @ ReactUpdates.js:201r @ ReactUpdateQueue.js:25enqueueSetState @ ReactUpdateQueue.js:210r.setState @ ReactComponent.js:64s.handleChange @ connect.js:301c @ createStore.js:186(anonymous function) @ middleware.js:22(anonymous function) @ index.js:28(anonymous function) @ index.js:14dispatch @ applyMiddleware.js:45(anonymous function) @ redux.js:57r @ runtime.js:62(anonymous function) @ runtime.js:336e.(anonymous function) @ runtime.js:95r @ app.bundle.js:1104(anonymous function) @ app.bundle.js:1104
```

**Comment by rahimrahman** [comment]
I'm having the same issue.  I have to refre

Metabase Version: 0.20.3 & 0.20.0
Browser: Chrome && Firefox
OS: Alpine
Database: Postgres

**Comment by rahimrahman** [comment]
@mattxbart the solution is quite simple (at least from my test table).  I had to hide some columns to get the number of columns to 32 or below and the issue went away.  The code below kinda give it away.  Kinda.

```
  crossfilter_arrayWiden = function(array, width) {
    var copy;
    switch (width) {
      case 16: copy = crossfilter_array16(array.length); break;
      case 32: copy = crossfilter_array32(array.length); break;
      default: throw new Error("invalid array width!");
    }
    copy.set(array);
    return copy;
  };
```

**Comment by mattxbart** [comment]
@rahimrahman yes, I came to the same conclusion. I know 32 is a lot of fields but max 64 would be cool if didn't have any serious consequences.

**Comment by salsakran** [comment]
crossfilter doesn't support more than 32 dimensions -- https://github.com/square/crossfilter/wiki/API-Reference#dimension

Is this on a table view or something else? This is a place where I'm not sure crossfilter is actually buying us anything as we're not re-filtering anything clientside. @tlrobinson do you know if there is a way to use dc.js for a table without treating every column as a crossfilter dimension.

**Comment by mattxbart** [comment]
This is a duplicate of #3407

**Comment by tlrobinson** [comment]
@mattxbart @rahimrahman We believe this has been fixed in `master`. If possible could you try it out and confirm.

You can build from `master`, use the `metabase/metabase-head:latest` Docker image, or this jar: https://9000-30203935-gh.circle-artifacts.com/6/home/ubuntu/metabase/target/uberjar/metabase.jar

**Comment by tlrobinson** [comment]
This is a duplicate of #3205. Please comment there if you can confirm this is or isn't fixed.

**Comment by rahimrahman** [comment]
@tlrobinson confirm that this issue has been fixed 0.21.0

**Comment by mattxbart** [comment]
I just confirmed it is fixed in 0.21.0, Thanks everyone!
